vcd/resource_vcd_nsxt_edgegateway_dns_test.go

```diff
@@ -0,0 +1,586 @@
+//go:build network || nsxt || ALL || functional
+
+package vcd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccVcdNsxtEdgegatewayDns(t *testing.T) {
+	preTestChecks(t)
+
+	// String map to fill the template
+	var params = StringMap{
+		"Org":                     testConfig.VCD.Org,
+		"NsxtVdc":                 testConfig.Nsxt.Vdc,
+		"VdcGroup":                testConfig.Nsxt.VdcGroup,
+		"VdcGroupEdgeGw":          testConfig.Nsxt.VdcGroupEdgeGateway,
+		"EdgeGw":                  testConfig.Nsxt.EdgeGateway,
+		"DnsConfig":               t.Name(),
+		"VdcGroupDnsConfig":       t.Name() + "vdcgroup",
+		"DefaultForwarderName":    t.Name() + "default",
+		"ConditionalForwardZone1": t.Name() + "conditional1",
+		"ConditionalForwardZone2": t.Name() + "conditional2",
+		"ServerIp1":               "1.1.1.1",
+		"ServerIp2":               "2.2.2.2",
+		"ServerIp3":               "3.3.3.3",
+		"ServerIp4":               "4.4.4.4",
+		"ServerIp5":               "5.5.5.5",
+		"DomainName1":             "example.org",
+		"DomainName2":             "testwebsite.org",
+		"DomainName3":             "nonexistent.nan",
+	}
+	testParamsNotEmpty(t, params)
+
+	params["FuncName"] = t.Name() + "step1"
+	configText1 := templateFill(testAccVcdNsxtEdgegatewayDnsStep1, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s\n", configText1)
+
+	params["FuncName"] = t.Name() + "step2"
+	configText2 := templateFill(testAccVcdNsxtEdgegatewayDnsStep2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 2: %s\n", configText2)
```

Comment:
Test in step 1 and step 2 are similar. They are needed in this framework, but it would be a waste in binary tests. Please add a skip statement for binary test in step 1 and remove it for step 2.

Reply:
Fixed.